_data/rules.yml

```diff
@@ -998,6 +998,66 @@ groups:
           - name: instaclustr/cassandra-exporter
             doc_url: https://github.com/instaclustr/cassandra-exporter
             rules:
+              - name: 'Cassandra Node is unavailable
+                description: 'Cassandra Node is unavailable - {{ $labels.cassandra_cluster }} {{ $labels.exported_endpoint }}'
+                query: 'sum(min_over_time(cassandra_endpoint_active{cassandra_cluster=~".*"}[1m])) by (cassandra_cluster,instance,exported_endpoint) < 1'
+                for: 2m
+                severity: critical
+              - name: 'Cassandra many compaction tasks are pending'
+                description: 'Many Cassandra compaction tasks are pending - {{ $labels.cassandra_cluster }}'
+                query: 'avg_over_time(cassandra_table_estimated_pending_compactions{cassandra_cluster=~".*"}[1m]) > 100'
+                for: 2m
+                severity: warning
+              - name: 'Cassandra commitlog pending tasks'
+                description: 'Cassandra commitlog pending tasks - {{ $labels.cassandra_cluster }}'
+                query: 'avg_over_time(cassandra_commit_log_pending_tasks{cassandra_cluster=~".*"}[1m]) > 15'
+                for: 2m
+                severity: warning
+              - name: 'Cassandra compaction executor blocked tasks'
+                description: 'Some Cassandra compaction executor tasks are blocked - {{ $labels.cassandra_cluster }}'
+                query: 'avg(avg_over_time(cassandra_thread_pool_blocked_tasks{cassandra_cluster=~".*",pool="CompactionExecutor"}[1m])) by (cassandra_cluster,instance) > 15'
+                for: 2m
+                severity: warning
+              - name: 'Cassandra flush writer blocked tasks'
+                description: 'Some Cassandra flush writer tasks are blocked - {{ $labels.cassandra_cluster }}'
+                query: 'avg(avg_over_time(cassandra_thread_pool_blocked_tasks{cassandra_cluster=~".*",pool="MemtableFlushWriter"}[1m])) by (cassandra_cluster,instance) > 15'
+                for: 2m
+                severity: warning
+              - name: 'Cassandra connection timeouts total'
+                description: 'Some connection between nodes are ending in timeout - {{ $labels.cassandra_cluster }}'
+                query: 'avg(avg_over_time(cassandra_client_request_timeouts_total{cassandra_cluster=~".*"}[1m])) by (cassandra_cluster,instance) > 5'
+                for: 2m
+                severity: critical
+              - name: 'Cassandra storage exceptions'
+                description: 'Something is going wrong with cassandra storage - {{ $labels.cassandra_cluster }}'
+                query: 'changes(cassandra_storage_exceptions_total{cassandra_cluster=~".*"}[1m]) > 1'
+                for: 2m
```

Comment:
In that case, I would remove the `for: 2m` statement. This might prevent some alarm to trigger.

But raising the threshold to `> 10` to prevent alarm fatigue. What do you think? Is `10` too high? Too low?

Reply:
Is it a critical issue or just a warning?

Reply:
I used threshold `> 1` and `critical` to be in sync with criteo exporter. 
In my setup I have them as warnings since I rely on a different monitoring for storage but I kept `1`. No false positives so far.